Assets/Scripts/Models/Area/World.cs

```diff
@@ -600,4 +631,36 @@ private void OnTileTypeChangedCallback(Tile t)
             tileGraph.RegenerateGraphAtTile(t.Down());
         }
     }
+
+    private void DetermineVisibility(Tile start)
+    {
+        Queue<Tile> roomsToVisit = new Queue<Tile>();
+
+        roomsToVisit.Enqueue(start);
+        Tile current;
+        do
+        {
+            current = roomsToVisit.Dequeue();
+
+            bool canEnterCurrent = current.IsEnterable() != Enterability.Never;
+
+            foreach (Tile neighbor in current.GetNeighbours(false, true, true))
+            {
+                if (neighbor == null || neighbor.CanSee)
```

Comment:
Am I correct in reading this as, if a tile somehow already has its CanSee set to true, this won't change it?

Reply:
That is correct. All tiles start out as can't see, and can have them turned on either here, or when initially set. This check prevents a stack overflow.